### PR TITLE
Feature/drag splits and filters between

### LINKED ISSUE #292: Droping (dimension) filter on split should add split and preserve filter
Worth noting is that would cause to "copy" dimension. We will get new split but filter will be still in place. Should we add some visual indicator of that?

### PR DESCRIPTION
**Pull request by adrianmroz**
Closes #292

### CODE REVIEW

File: src/client/components/filter-tile/filter-tile.tsx [hunk 1]
```diff
@@ -251,6 +251,9 @@ export class FilterTile extends React.Component<FilterTileProps, FilterTileState
     const { essence: { filter } } = this.props;
     const dimension = DragManager.draggingDimension();
     if (dimension) return !filter.getClauseForDimension(dimension);
+    if (DragManager.isDraggingSplit()) {
+      return !filter.clauseForReference(DragManager.draggingSplit().reference);
```
Comment (mkuthan) [inline src/client/components/filter-tile/filter-tile.tsx]: I would put all the logic in the DragManager, eg. DragManager.canDropFilterIntoSplit(filter.reference). But I'm not self-convinced.

File: src/client/components/filter-tile/filter-tile.tsx [hunk 2]
```diff
@@ -274,54 +277,51 @@ export class FilterTile extends React.Component<FilterTileProps, FilterTileState
     this.setState({ dragPosition: null });
   }
 
+  draggingDimension(): Dimension {
+    const { essence: { dataCube } } = this.props;
+    if (DragManager.isDraggingSplit()) {
```
Comment (mkuthan) [inline src/client/components/filter-tile/filter-tile.tsx]: Again, this logic should be fully encapsulated in DragManager, it is responsible for returning dragging dimension (if any).

File: src/client/components/filter-tile/filter-tile.tsx [hunk 3]
```diff
@@ -274,54 +277,51 @@ export class FilterTile extends React.Component<FilterTileProps, FilterTileState
...
-      if (dragPosition.replace !== null) {
-        const targetClause = filter.clauses.get(dragPosition.replace);
-        tryingToReplaceTime = targetClause && targetClause.reference === dataCube.getTimeDimension().name;
-      }
-      if (dragPosition && !tryingToReplaceTime) {
-        this.addDummy(dimension, dragPosition);
-      }
+
+    if (DragManager.isDraggingFilter()) {
+      this.dropFilter(dragPosition);
```
Comment (mkuthan) [inline src/client/components/filter-tile/filter-tile.tsx]: Again, isDraggingFilter condition could be checked internally by DragManger.

File: src/client/components/filter-tile/filter-tile.tsx (hunk 1 above)
Comment (adrianmroz) [reply src/client/components/filter-tile/filter-tile.tsx]: So we will get three functions: `canDropIntoSplit :: splits -> boolean`, `canDropIntoSeries :: series -> boolean`, `canDropIntoFilter :: filter -> boolean`. I can vouch for that ...

File: src/client/components/filter-tile/filter-tile.tsx (hunk 2 above)
Comment (adrianmroz) [reply src/client/components/filter-tile/filter-tile.tsx]: This is possible, but we will get three variants of this, for each TileMenu. Sometimes DragManager will search dimension from it's split, sometimes from it's filter.

File: src/client/components/filter-tile/filter-tile.tsx (hunk 3 above)
Comment (adrianmroz) [reply src/client/components/filter-tile/filter-tile.tsx]: I don't get that. I have one behaviour for dropping Filter and different for dropping Dimension. I think it should be placed there, but that way I need this condition to decide what to do.

File: src/client/components/filter-tile/filter-tile.tsx (hunk 1 above)
Comment (adrianmroz) [reply src/client/components/filter-tile/filter-tile.tsx]: Sometimes, we need to check dimension in dataCube, `DragManager` will start be aware of essence and dataCube. That seems wrong.

**Pull request merged**